.goreleaser.yml

```diff
@@ -1,96 +1,61 @@
 # Copyright (c) HashiCorp, Inc.
 # SPDX-License-Identifier: MPL-2.0
 
-# This is an example goreleaser.yaml file with some defaults.
-# Make sure to check the documentation at http://goreleaser.com
-version: 1
-
-env:
-  - CGO_ENABLED=0
-
-before:
-  hooks:
-    # We strongly recommend running tests to catch any regression before release.
-    # Even though, this an optional step.
-    - go test ./...
-    # Check plugin compatibility with required version of the Packer SDK
-    - make plugin-check
+# See http://goreleaser.com
+# version: 1
 
 builds:
-  # A separated build to run the packer-plugins-check only once for a linux_amd64 binary
-  - id: plugin-check
-    mod_timestamp: "{{ .CommitTimestamp }}"
-    flags:
-      - -trimpath #removes all file system paths from the compiled executable
-    ldflags:
-      - "-s -w -X {{ .ModulePath }}/version.Version={{.Version}} -X {{ .ModulePath }}/version.VersionPrerelease= "
-    goos:
-      - linux
-    goarch:
-      - amd64
-    binary: "{{ .ProjectName }}_v{{ .Version }}_{{ .Env.API_VERSION }}_{{ .Os }}_{{ .Arch }}"
-  
-  - id: linux-builds
+  - id: build
     mod_timestamp: "{{ .CommitTimestamp }}"
+    env:
+      - CGO_ENABLED=0
     flags:
-      - -trimpath #removes all file system paths from the compiled executable
+      - -trimpath
     ldflags:
-      - "-s -w -X {{ .ModulePath }}/version.Version={{.Version}} -X {{ .ModulePath }}/version.VersionPrerelease= "
-    goos:
-      - linux
-    goarch:
-      - amd64
-      - "386"
-      - arm
-      - arm64
-    ignore:
-      - goos: linux
-        goarch: amd64
-    binary: "{{ .ProjectName }}_v{{ .Version }}_{{ .Env.API_VERSION }}_{{ .Os }}_{{ .Arch }}"
+      - -s
+      - -w
+      - -X {{ .ModulePath }}/version.Version={{ .RawVersion }}
+      - -X {{ .ModulePath }}/version.VersionPrerelease={{ .Prerelease }}
```

Comment:
Related to https://github.com/hetznercloud/packer-plugin-hcloud/blob/bc350955f001ed9fda34adb3267f70ff2d67fde8/version/version.go#L9-L21

We need to decide what the single source of truth is for our version, and plan accordingly.

Reply:
Because right now we have the commit version but still override it through the linker flags based on the tag goreleaser is executed with?

I think using the committed `Version` is best, while also using the prelease override through Goreleaser to be able to differentiate official builds from local ones (and test suites). We might even be able to adapt this pattern for some other integrations where we have the same issue (dev/test vs official build)

Reply:
Updates from private chat: While the version looks ugly if we use `release-please` to create actual prereleases, the resulting version (`1.2.0-rc0-dev`) is still a valid [SemVer](https://semver.org/#spec-item-9)